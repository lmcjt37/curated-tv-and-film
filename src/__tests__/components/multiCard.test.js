import React from 'react';
import { render, screen } from '@testing-library/react';

import MultiCard from '../../components/multiCard';

const itemMockOne = {
  type: 'tv_show',
  title: 'The Walking Dead',
  genre: ['Drama', 'Horror', 'Action'],
  content: [
    {
      description:
        'Following on from where Rick Grimes wakes up from a coma and tries to find his family. He gets to Atlanta, surrounded by walkers, he is thrown from his horse and ends up in a tank. This is where someone on the walkie talkie, Glenn, helps him out of the tight situation.',
      season: 1,
      episode: 2,
      episode_title: 'Guts',
      imdb: 'https://www.imdb.com/title/tt1628064/?ref_=ttep_ep2',
      url: 'https://www.youtube.com/watch?v=ZxmWhlidAZ8',
      thumbnail: './assets/images/walking-dead.jpg'
    }
  ]
};

const itemMockTwo = {
  type: 'tv_show',
  title: 'The Walking Dead',
  genre: ['Drama', 'Horror', 'Action'],
  content: [
    {
      description:
        'Following on from where Rick Grimes wakes up from a coma and tries to find his family. He gets to Atlanta, surrounded by walkers, he is thrown from his horse and ends up in a tank. This is where someone on the walkie talkie, Glenn, helps him out of the tight situation.',
      season: 1,
      episode: 2,
      episode_title: 'Guts',
      imdb: 'https://www.imdb.com/title/tt1628064/?ref_=ttep_ep2',
      url: 'https://www.youtube.com/watch?v=ZxmWhlidAZ8',
      thumbnail: './assets/images/walking-dead.jpg'
    },
    {
      description:
        'The Saviours have captured Rick and the group. They kneel at the mercy of Negan who makes an example of 2 of the groups members. Not for the faint of heart.',
      season: 7,
      episode: 1,
      episode_title: "The Day Will Come When You Won't Be",
      imdb: 'https://www.imdb.com/title/tt5164772/?ref_=ttep_ep1',
      url: 'https://www.youtube.com/watch?v=d3LDG5CCOPQ',
      thumbnail: './assets/images/walking-dead.jpg'
    }
  ]
};

it('render all initial child components with one items', () => {
  render(<MultiCard {...itemMockOne} />);

  expect(screen.getByTestId('multi-card')).toBeTruthy();
  expect(screen.getByTestId('multi-card-title')).toBeTruthy();
  expect(screen.getByTestId('multi-card-chip-container')).toBeTruthy();
  expect(
    screen.getAllByTestId('multi-card-chip-container-item').length
  ).toEqual(3);

  expect(screen.getAllByTestId('multi-card-item').length).toEqual(1);
  expect(screen.getAllByTestId('multi-card-media').length).toEqual(1);
  expect(screen.getAllByTestId('multi-card-image').length).toEqual(1);
  expect(screen.getAllByTestId('multi-card-content').length).toEqual(1);
  expect(screen.getAllByTestId('multi-card-subtitle-1').length).toEqual(1);
  expect(screen.getAllByTestId('multi-card-subtitle-2').length).toEqual(1);
  expect(screen.getAllByTestId('multi-card-description').length).toEqual(1);
  expect(screen.getAllByTestId('multi-card-actions').length).toEqual(1);
  expect(screen.getAllByTestId('multi-card-action-imdb').length).toEqual(1);
  expect(screen.getAllByTestId('multi-card-action-watch').length).toEqual(1);
});

it('render all initial child components with two items', () => {
  render(<MultiCard {...itemMockTwo} />);

  expect(screen.getByTestId('multi-card')).toBeTruthy();
  expect(screen.getByTestId('multi-card-title')).toBeTruthy();
  expect(screen.getByTestId('multi-card-chip-container')).toBeTruthy();
  expect(
    screen.getAllByTestId('multi-card-chip-container-item').length
  ).toEqual(3);

  expect(screen.getAllByTestId('multi-card-item').length).toEqual(2);
  expect(screen.getAllByTestId('multi-card-media').length).toEqual(2);
  expect(screen.getAllByTestId('multi-card-image').length).toEqual(2);
  expect(screen.getAllByTestId('multi-card-content').length).toEqual(2);
  expect(screen.getAllByTestId('multi-card-subtitle-1').length).toEqual(2);
  expect(screen.getAllByTestId('multi-card-subtitle-2').length).toEqual(2);
  expect(screen.getAllByTestId('multi-card-description').length).toEqual(2);
  expect(screen.getAllByTestId('multi-card-actions').length).toEqual(2);
  expect(screen.getAllByTestId('multi-card-action-imdb').length).toEqual(2);
  expect(screen.getAllByTestId('multi-card-action-watch').length).toEqual(2);
});

it('snapshot of initial component with multiple items', () => {
  const { asFragment } = render(<MultiCard {...itemMockTwo} />);
  expect(asFragment).toMatchSnapshot();
});
