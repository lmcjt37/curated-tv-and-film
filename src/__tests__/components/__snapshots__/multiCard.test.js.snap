// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot of initial component with multiple items 1`] = `
<div
  class="MuiPaper-root MuiCard-root MultiCard-card-1 MuiPaper-elevation1 MuiPaper-rounded"
>
  <div
    class="MuiCardHeader-root"
  >
    <div
      class="MuiCardHeader-content"
    >
      <h2
        class="MuiTypography-root MuiCardHeader-title MuiTypography-h6 MuiTypography-displayBlock"
      >
        The Walking Dead
      </h2>
    </div>
  </div>
  <div
    class="MuiCardContent-root"
  >
    <div
      class="MuiChip-root MultiCard-chip-6"
    >
      <span
        class="MuiChip-label"
      >
        Drama
      </span>
    </div>
    <div
      class="MuiChip-root MultiCard-chip-6"
    >
      <span
        class="MuiChip-label"
      >
        Horror
      </span>
    </div>
    <div
      class="MuiChip-root MultiCard-chip-6"
    >
      <span
        class="MuiChip-label"
      >
        Action
      </span>
    </div>
  </div>
  <div
    class="MultiCard-innerContainer-2"
  >
    <div
      class="MuiCardMedia-root MultiCard-media-3"
      style="background-image:url(\\"./assets/images/walking-dead.jpg\\")"
    />
    <img
      class="MultiCard-thumbnail-5"
      src="./assets/images/walking-dead.jpg"
    />
    <div
      class="MuiCardContent-root"
    >
      <div>
        <div>
          <h2
            class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-gutterBottom"
          >
            Guts
          </h2>
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary MuiTypography-gutterBottom"
          >
            Season: 1 | Episode: 2
          </p>
        </div>
        <p
          class="MuiTypography-root MultiCard-description-4 MuiTypography-body1"
        >
          Following on from where Rick Grimes wakes up from a coma and tries to find his family. He gets to Atlanta, surrounded by walkers, he is thrown from his horse and ends up in a tank. This is where someone on the walkie talkie, Glenn, helps him out of the tight situation.
        </p>
      </div>
      <div
        class="MuiCardActions-root MultiCard-actions-10 MuiCardActions-spacing"
      >
        <a
          aria-disabled="false"
          aria-label="IMDB"
          class="MuiButtonBase-root MuiFab-root MultiCard-fab-7 MuiFab-extended MuiFab-primary"
          href="https://www.imdb.com/title/tt1628064/?ref_=ttep_ep2"
          rel="noopener noreferrer"
          tabindex="0"
          target="_blank"
        >
          <span
            class="MuiFab-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MultiCard-localPlayIcon-9"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M20 12c0-1.1.9-2 2-2V6c0-1.1-.9-2-2-2H4c-1.1 0-1.99.9-1.99 2v4c1.1 0 1.99.9 1.99 2s-.89 2-2 2v4c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2v-4c-1.1 0-2-.9-2-2zm-4.42 4.8L12 14.5l-3.58 2.3 1.08-4.12-3.29-2.69 4.24-.25L12 5.8l1.54 3.95 4.24.25-3.29 2.69 1.09 4.11z"
              />
            </svg>
            IMDB
          </span>
        </a>
        <a
          aria-disabled="false"
          aria-label="Watch"
          class="MuiButtonBase-root MuiFab-root MultiCard-fab-7 MuiFab-extended MuiFab-primary"
          href="https://www.youtube.com/watch?v=ZxmWhlidAZ8"
          rel="noopener noreferrer"
          tabindex="0"
          target="_blank"
        >
          <span
            class="MuiFab-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MultiCard-playCircleOutlineIcon-8"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M10 16.5l6-4.5-6-4.5v9zM12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
              />
            </svg>
            Watch
          </span>
        </a>
      </div>
    </div>
  </div>
  <div
    class="MultiCard-innerContainer-2"
  >
    <div
      class="MuiCardMedia-root MultiCard-media-3"
      style="background-image:url(\\"./assets/images/walking-dead.jpg\\")"
    />
    <img
      class="MultiCard-thumbnail-5"
      src="./assets/images/walking-dead.jpg"
    />
    <div
      class="MuiCardContent-root"
    >
      <div>
        <div>
          <h2
            class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-gutterBottom"
          >
            The Day Will Come When You Won't Be
          </h2>
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary MuiTypography-gutterBottom"
          >
            Season: 7 | Episode: 1
          </p>
        </div>
        <p
          class="MuiTypography-root MultiCard-description-4 MuiTypography-body1"
        >
          The Saviours have captured Rick and the group. They kneel at the mercy of Negan who makes an example of 2 of the groups members. Not for the faint of heart.
        </p>
      </div>
      <div
        class="MuiCardActions-root MultiCard-actions-10 MuiCardActions-spacing"
      >
        <a
          aria-disabled="false"
          aria-label="IMDB"
          class="MuiButtonBase-root MuiFab-root MultiCard-fab-7 MuiFab-extended MuiFab-primary"
          href="https://www.imdb.com/title/tt5164772/?ref_=ttep_ep1"
          rel="noopener noreferrer"
          tabindex="0"
          target="_blank"
        >
          <span
            class="MuiFab-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MultiCard-localPlayIcon-9"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M20 12c0-1.1.9-2 2-2V6c0-1.1-.9-2-2-2H4c-1.1 0-1.99.9-1.99 2v4c1.1 0 1.99.9 1.99 2s-.89 2-2 2v4c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2v-4c-1.1 0-2-.9-2-2zm-4.42 4.8L12 14.5l-3.58 2.3 1.08-4.12-3.29-2.69 4.24-.25L12 5.8l1.54 3.95 4.24.25-3.29 2.69 1.09 4.11z"
              />
            </svg>
            IMDB
          </span>
        </a>
        <a
          aria-disabled="false"
          aria-label="Watch"
          class="MuiButtonBase-root MuiFab-root MultiCard-fab-7 MuiFab-extended MuiFab-primary"
          href="https://www.youtube.com/watch?v=d3LDG5CCOPQ"
          rel="noopener noreferrer"
          tabindex="0"
          target="_blank"
        >
          <span
            class="MuiFab-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MultiCard-playCircleOutlineIcon-8"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M10 16.5l6-4.5-6-4.5v9zM12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
              />
            </svg>
            Watch
          </span>
        </a>
      </div>
    </div>
  </div>
</div>
`;
